### DISCUSSION TITLE
Some Questions about the Panda Result

### DISCUSSION THREAD

**Discussion by lijinw** [netZoo/netZooR#243] [Q&A] [answered]
Hi,

My name is Lijin Wang, and I’m a research assistant at National University of Singapore. Thanks for developing the wonderful packages panda. I finished building adipose regulatory network based on the Panda package. But I have a question about meaning of the value show up in the regNet network. In the tutorial(https://ddec1-0-en-ctp.trendmicro.com:443/wis/clicktime/v1/query?url=https%3a%2f%2fnetzoo.github.io%2fnetZooR%2farticles%2fApplicationinGTExData%5fDeborahW.html&umid=41eabb56-d530-4161-9a29-5e141b3ff6c6&auth=3eee6d57317a38631073652579c10dc620ca2b41-dc6ef46f8428978b2652a4ff0c896ef173bca68e) , the value means the edge weights which representing the "likelihood" that a transcription factor binds the promotor of and regulates the expression of a gene. I want to make a plot with the transcription factor that most likely to binds the promotor of and regulates the expression of a gene. So I wonder if negative likelihood means that the transcription factor not likely to binds the promotor of and regulates the expression of a gene, or it has any other meaning? Should I consider the absolute value when I selected TF or just remove those with negative value?  Thanks a lot!
 
Best,
Lijin

**Comment by marouenbg** [comment]
Hi @lijinw ,

Thank you for posting on discussions. Indeed, negative PANDA edge weight mean that the specific edge is not likely to bind the promoter of the gene and regulate it. However, the sign does not have any meaning about direction of regulation for example, that's why it is not recommended to do the absolute value on edge weights. For your network plot, you can select the top n largest edges in your network and plot them. Let me know if I can help in any way!
Marouen

**Reply by lijinw** [reply]
Hi @marouenbg 

Thanks for your reply! That was really helpful. Now I'm trying to use panda to build network for each of the subgroup, and using the result to conduct the alpaca. But I found my computer was breakdown when runing it, I guess this is because the motif file was too big. I download the motif and ppi file from grand dataset, after mapping to the my gene list. I still have 8939364 rows in the motif file  and  84388 rows in the ppi file. I wonder if there is anyway to narrow the motif and ppi file? Thanks again.

**Comment by marouenbg** [comment]
Hi @lijinw ,
Yes, sometimes you might need more memory to run on large input files.
Try running with mode='intersection' this should reduce the input files to their intersection.
Which motif file are you using?

**Reply by lijinw** [reply]
Hi @marouenbg,

Thanks for your reply. Now I have built separate network for obesity group and lean group by panda. I dowload the motif file from Adipose visceral networks at GRAND dataset (the motif file is quite big! ), and I select those motif and genes only show up in my gene list. After that, i have used louvain to detect the commnuity in each network, I wonder do I need to remove those with negative likelihood before I start detect the community? Also, the input file that I used in alpaca is only one commnuity or I can put all of the commnuity together and running it? I would appreciate if you could answer those questions.

Best,
Lijin

**Comment by marouenbg** [comment, marked as answer]
Hi @lijinw ,

Your approach is quite sensible !
For community detection, try using CONDOR (https://netzoo.github.io/zooanimals/condor/). 

For Louvain, i used it a while back and I forgot if it requires all positive numbers. At this point, I think it is a matter of trial and error if you want include the negative edges or remove them.

As for ALPACA, it takes as input 2 networks (case and control) not a community assignment of networks. Then it will run community detection to find differential communities.

Hope that helps!

**Comment by lijinw** [comment]
Hi @marouenbg,

Thanks for your reply! That helps a lot!!! I read the paper about alpaca (https://www.nature.com/articles/s41540-018-0052-5), then I thought condor or louvain should be one step before alpaca. Thanks for let me know that I can start ALPACA by just inputing the 2 networks. I have finished this part. Thanks again!

**Reply by marouenbg** [reply]
Yes you can run ALPACA directly.
